fix some bugs in Makefile

CohenCyril opened the pull request.

amahboubi commented:
I do not know what to do with this PR. I am unable to review without at least a small description of the pb.
@gares may be more relevant as a reviewer?

gares commented:
I'd be happy if the code could call
```
$ ~/COQ/coq/bin/coqtop -print-version
8.9+alpha 4.06.0
```
and "parse" that string, that was made easier to parse (the second version is the ocaml compiler)

CohenCyril commented:
@gares actually `-print-version` seems to be unavailable for coq 8.6, so the modification you suggest actually implies dropping support...

CohenCyril commented:
@gares I'd rather push this fix first, and later make another PR taking your remark into account and drop 8.6 when there is a consensus. I left a comment in the `Makefile.common` with the instructions.

gares commented:
I guess I added --print-version on 8.6, and then the more standard -print-version was also added. Could you try that? (I don't have a 8.6 around)

CohenCyril commented:
> I guess I added --print-version on 8.6, and then the more standard -print-version was also added. Could you try that? (I don't have a 8.6 around)

With nix, coq 8.6 was around in a few seconds :heart:

gares commented:
I'm a bit lost. Is the change I propose doable on 8 6 too or not? If not I shall merge as is. Else I should wait for a fix.

CohenCyril commented:
@gares the fix was made (unless there is more to fix than I thought) and it passed.

gares commented on mathcomp/Makefile.common:
```diff
@@ -64,7 +76,7 @@ install: __always__ Makefile.coq
 .PHONY: count
 
-COQFILES := $(shell grep '.v$$' Make)
+COQFILES = $(shell grep '.v$$' $(COQPROJECT))
```
@CohenCyril @SkySkimmer wrt #238 I think this line is fishy. `$(COQPROJECT)` points to `_CoqProject` that, IIRC, only contains `-R` directives but not the list of files.

SkySkimmer replied:
I believe it points to https://github.com/math-comp/math-comp/blob/d6dc5741ba44808e5f2f01a238d972ec2c11737f/mathcomp/fingroup/Make (which does have v files) as per https://github.com/math-comp/math-comp/blob/d6dc5741ba44808e5f2f01a238d972ec2c11737f/mathcomp/fingroup/Makefile#L3

CohenCyril replied:
> [...] as per
> 
> [math-comp/mathcomp/fingroup/Makefile](https://github.com/math-comp/math-comp/blob/d6dc5741ba44808e5f2f01a238d972ec2c11737f/mathcomp/fingroup/Makefile#L3)
> 
> Line 3 in [d6dc574](/math-comp/math-comp/commit/d6dc5741ba44808e5f2f01a238d972ec2c11737f)
> 
>  COQPROJECT="Make"

Yes indeed, but... could the double quotes be guilty here?

SkySkimmer replied:
No, it gets sent to the shell which will remove them.

SkySkimmer commented on mathcomp/Makefile.common:
```diff
@@ -23,14 +23,26 @@ VERBOSE?=V
 .PHONY: pre-makefile
 
-Makefile.coq: pre-makefile Makefile
+Makefile.coq: pre-makefile $(COQPROJECT) Makefile
```
I think the issue is the double quotes, although they are a problem on this line not the grep line.
